slatedb/src/wal_buffer.rs

```
@@ -228,9 +231,14 @@ impl WalBufferManager {
             flush_tx
                 .as_ref()
                 .expect("flush_tx not initialized, please call start_background first.")
-                .send(WalFlushWork { result_tx: None })
-                .await
-                .map_err(|_| SlateDBError::BackgroundTaskShutdown)?;
+                .try_send(WalFlushWork { result_tx: None })
+                .or_else(|err| match err {
+                    TrySendError::Full(_) => {
```

Should we check the error here? Seems like it might mask other errors, which would be misleading.

---

just jumped into the code, i found that `TrySendError::Full(_)` does not contain an inner error, but the original message we sent to the channel 😲

```
    pub fn try_send(&self, message: T) -> Result<(), TrySendError<T>> // <--- the T in TrySendError is the same type of message
```

```
/// This enumeration is the list of the possible error outcomes for the
/// [`try_send`](super::Sender::try_send) method.
#[derive(PartialEq, Eq, Clone, Copy)]
pub enum TrySendError<T> {
    /// The data could not be sent on the channel because the channel is
    /// currently full and sending would require blocking.
    Full(T),

    /// The receive half of the channel was explicitly closed or has been
    /// dropped.
    Closed(T),
}
```

we could check these two cases about `Full` or `Closed`, but there's no more embeded error inside this TrySendError imo.

---

Gotcha! Makes sense. tokio channels are very strange...